apollo-federation/src/merge.rs

```diff
@@ -600,23 +647,128 @@ impl Merger {
     }
 
     fn merge_scalar_type(
-        &self,
+        &mut self,
         types: &mut IndexMap<Name, ExtendedType>,
+        directive_names: &DirectiveNames,
         subgraph_name: Name,
         scalar_name: NamedType,
         ty: &Node<ScalarType>,
     ) {
         let existing_type = types
             .entry(scalar_name.clone())
-            .or_insert(copy_scalar_type(scalar_name, ty));
+            .or_insert(copy_scalar_type(scalar_name.clone(), ty));
```

Comment:
Is this clone needed?

Reply:
nope, this is leftover after refactoring